### DISCUSSION TITLE
Shapes and uploaded images are not interactive in React

### DISCUSSION THREAD

**Discussion by pini85** [fabricjs/fabric.js#8112] [Q&A] [answered]
## Version
5.0.0

## Test Case
- [ ] [Browser Issue Template](https://codesandbox.io/s/epic-shirley-5l3u9p?file=/src/App.js)

## Information about environment
Browser
All Browsers

## Steps To Reproduce

1.  Upload an image or create a shape and try to interactive with it.

## Expected Behavior
To interactive with elements

## Actual Behavior
Cannot interactive with the elements

[demo](https://codesandbox.io/s/epic-shirley-5l3u9p?file=/src/App.js)

video:
https://user-images.githubusercontent.com/40143951/182140326-a5e32b96-40e9-46c5-b1d2-f89e7446764c.mov

**Comment by ShaMan123** [comment]
try downgrading to react 17
v18 had some bugs reported with fabric

**Comment by asukadev** [comment, marked as answer]
Hey !

Because you are using StrictMode on your whole App,your components are getting rerendered multiple times, and because you don't manage the life cycle flow of each render correctly, your canvas is getting wrapped multiple times. 

Personally, I'd rather store the canvas inside a useRef rather than a useState and dont forget to dispose the canvas on the unmount event of the component.

I modified your codepen, hope you can easily understand 👍 

[https://codesandbox.io/s/focused-clarke-tvkjs6?file=/src/App.js](https://codesandbox.io/s/focused-clarke-tvkjs6?file=/src/App.js)
